data/mapcss-mapping.csv

```diff
@@ -1660,3 +1660,10 @@ craft|grinding_mill;1584;
 leisure|outdoor_seating;1585;
 railway|station|light_rail|london;[railway=station][station=light_rail][city=london];;name;int_name;1586;
 railway|station|light_rail|porto;[railway=station][station=light_rail][city=porto];;name;int_name;1587;
+amenity|parking|fee|underground;[amenity=parking][location=underground][fee],[amenity=parking][parking=underground][fee];;name;int_name;1588;
+# The following two types are needed to override both amenity|parking|underground and amenity|parking|private
+amenity|parking|underground|private;[amenity=parking][location=underground][access=private],[amenity=parking][parking=underground][access=private];;name;int_name;1589;
+amenity|parking|private|underground;[amenity=parking][location=underground][access=private],[amenity=parking][parking=underground][access=private];;name;int_name;1590;
+# The following two types are needed to override both amenity|parking|street_side and amenity|parking|private
```

Comment:
Mmm, if we'd change a basic street_side mapping to `amenity|parking|street_side;[amenity=parking][parking=street_side][!fee][!private]` then it won't match here and we will not have to override it and will end up with a single `amenity|parking|private|street_side` type in a feature instead of two. Right?

Reply:
I don't think that works. It should be `amenity|parking|street_side;[amenity=parking][parking=street_side][!fee][access!=private]` but it doesn't work either.

Reply:
Hmm, 🤔 I just tested again with `amenity|parking|street_side;[amenity=parking][parking=street_side][!fee][access!=private]` and it seems to be working? I will do some more testing and come back

Reply:
ah sorry, you're right `!private` will not work and `access!=private` is not supposed to work either :)
so your original variant looks like the only working option indeed!
sorry for the confusion

Reply:
It does indeed seem to be working. Here is without `[access!=private]`:

![Captura de pantalla 2023-10-03 a las 20 12 04](https://github.com/organicmaps/organicmaps/assets/47610359/8dbccc0b-42cd-4b15-a3c4-b7571028651b)

And here is with `[access!=private]`:

![Captura de pantalla 2023-10-03 a las 20 23 17](https://github.com/organicmaps/organicmaps/assets/47610359/99c82e1a-3184-4576-8eb1-e47b7f8e023b)

Reply:
Did you check that regular `amenity-parking-underground` didn't disappear?
I guess adding `[access!=private]` will have the same effect as e.g. adding `[dont=match]` so it won't match anything.. 

I would be surprised if it works for all cases as its written though - it would mean I have to re-read this matching code and update docs/comments :)

Reply:
> Did you check that regular `amenity-parking-underground` didn't disappear? I guess adding `[access!=private]` will have the same effect as e.g. adding `[dont=match]` so it won't match anything..

You are right! It is not matching anything. Regular `amenity-parking-underground` are disappearing. 😂

Reverting back to how it was before.